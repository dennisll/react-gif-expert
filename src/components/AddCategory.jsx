import PropTypes from "prop-types";
import { useState } from "react";


export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChance = ({ target }) => {
    setInputValue(target.value);
  }

  const onSubmit = (event) => {
    event.preventDefault();
    if(inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue(''); 
  }
// onChange={(event) => onInputChance(event)} esto es lo mismo que 
  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChance}
      />
    </form>

  )
}

AddCategory.propTypes ={
  onNewCategory: PropTypes.func.isRequired
}
