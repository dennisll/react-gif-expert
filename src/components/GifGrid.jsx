
import { GifItem } from './GifItem';
import { userFetchGifs } from '../hooks/userFetchGifs';


export const GifGrid = ({ category }) => {
  // este es un hook personalizado, que contiene en 
  //su interior varios hooks de react
  const { images, isLoading } = userFetchGifs(category);

  console.log(isLoading);

  //{...image} es equivalente a enviar todas las propiedades de la imagen como props

  return (
    <>
      <h3>{category}</h3>

      {
        isLoading && <h2>...Cargando</h2>
      }

      {
      /*   {
        isLoading ? <h2>...Cargando</h2> : null
      } */
      }

      <div>
        {images.map(image => (
          <GifItem
            key={image.id}
            {...image}
          />
        ))}
      </div>


    </>
  )
}

