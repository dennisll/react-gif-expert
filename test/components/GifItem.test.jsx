import {expect, describe, test} from '@jest/globals';
import { render, screen} from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';


describe('Test GifItem component', () => {
  
    const title = 'title';
    const url = 'https://kinsta.com/es/wp-content/uploads/sites/8/2021/02/what-is-a-url-1024x512.jpeg'
      

    test('should be match with snapshot', () => {

        const {container} = render( <GifItem 
            title={ title }
            url = { url }
            />);

        expect(container).toMatchSnapshot();

    });

    test('debe mostrar la imagen con el alt y el source indicado', () => {
      
        render( <GifItem 
            title={ title }
            url = { url }
            />);
            const { alt, src} = screen.getByRole('img');
            expect(alt).toBe(title);
            expect(src).toBe(url);
    });


    test('debe mostrar el title en el componente', () => {
      
        render( <GifItem 
            title={ title }
            url = { url }
            />);
            expect(screen.getByText(title)).toBeTruthy();
           
    });
    
});
