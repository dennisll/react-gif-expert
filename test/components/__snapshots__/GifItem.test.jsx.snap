// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem component should be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://kinsta.com/es/wp-content/uploads/sites/8/2021/02/what-is-a-url-1024x512.jpeg"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
